examples/step-49/doc/results.dox

```diff
@@ -225,158 +239,109 @@ void create_3d_grid (Triangulation<3> &triangulation)
-
-  // Then loop over all faces of the domain and, if for the position
-  // of the center of a face the following holds then set boundary
-  // indicators:
-  // - if y>3 and z<=2.5 or z>=5 then use boundary indicator 8
-  // - if y>3 and 2.5<=z<=5 then use boundary indicator 9
-  typename Triangulation<3>::active_cell_iterator
-    cell = triangulation.begin_active(),
-    endc = triangulation.end();
-  for (; cell!=endc; ++cell)
-    for (unsigned int f=0;
-	 f < GeometryInfo<3>::faces_per_cell;
-	 ++f)
-      {
-	const Point<3> face_center = cell->face(f)->center();
-
-	if (cell->face(f)->at_boundary())
-	  {
-	    if ((face_center[2] <= 2.5 || face_center[2] >= 5) &&
-	        face_center[1] >= 3)
-	      cell->face(f)->set_boundary_id(8);
-
-	    if (face_center[2] >= 2.5 &&
-	        face_center[2] <= 5
-	        && face_center[1] >= 3)
-	      cell->face(f)->set_boundary_id(9);
-	  }
-      }
+  create_3d_grid(triangulation);
+  triangulation.set_manifold (cylinder_id, cylinder);
+
+  for (auto cell : triangulation.active_cell_iterators())
+    if (cell->center()[1] >= 3.0)
+      cell->set_all_manifold_ids(cylinder_id);
 
-  // Then refine the mesh once
   triangulation.refine_global (1);
 @endcode
 
 With this code, we get a mesh that looks like this:
 
-<img src="https://www.dealii.org/images/steps/developer/step-49.yuhan.3.png" alt="">
+<img src="https://www.dealii.org/images/steps/developer/step-49.yuhan.9.png"
+     alt="" width="400" height="355">
 
-This is clearly not correct: The new vertices that have been entered at
-mid-edge and mid-face points are not where they should have been. Upon some
-reflection, it turns out that while the radii of the cylinders are correct,
-the axes of the two cylinder objects should not have been along coordinate
-axes but shifted. This can be corrected by creating them as follows, the
-two points given as arguments indicating the direction and a point on the
-axis:
+This change fixes the boundary but creates a new problem: the cells adjacent to
+the cylinder's axis are badly distorted. We should use Cartesian coordinates for
+calculations on these central cells to avoid this issue. The cells along the
+center line all have a face that touches the line $(0, 3, z)$ so, to implement
+this, we go back and overwrite the <code>manifold_id</code>s on these cells to
+be zero (which is the default):
 
 @code
-  static const CylinderBoundary<3> inner_cylinder (inner_radius,
-                                                   Point<3>(0,0,1),
-                                                   Point<3>(0,3,0));
-  static const CylinderBoundary<3> outer_cylinder (outer_radius,
-                                                   Point<3>(0,0,1),
-                                                   Point<3>(0,3,0));
-  triangulation.set_boundary (9, outer_cylinder);
-@endcode
+  const Tensor<1, 3> axis {{0.0, 0.0, 1.0}};
+  const Point<3> axial_point {0, 3.0, 0.0};
```

Comment:
The different number of curly brackets looks inconsistent. Isn't a single pair sufficient?